surch/organization.py

```diff
@@ -19,199 +19,200 @@
-                              remove_cloned_dir=False,
-                              organization_flag=True):
-        conf_vars = utils.read_config_file(
-            config_file=config_file,
-            print_result=print_result,
-            verbose=verbose,
-            remove_cloned_dir=remove_cloned_dir,
-            organization_flag=organization_flag)
+                              is_organization=True,
+                              remove_cloned_dir=False):
+        conf_vars = utils.read_config_file(verbose=verbose,
+                                           config_file=config_file,
+                                           print_result=print_result,
+                                           is_organization=is_organization,
+                                           remove_cloned_dir=remove_cloned_dir)
         return cls(**conf_vars)
 
-    def get_github_repo_list(
-            self,
-            url_type='clone_url',
-            repository_type='public',
-            repository_per_page=100):
-        """This method get from GitHub the git url list for cloning
-        """
-        lgr.info(
-            'Retrieving list of repositories for the {0}...'.format(
-                self.object_type))
-        auth = (self.git_user, self.git_password) if self.auth else False
-        repository_data = \
-            requests.get(
-                ORG_DETAILS_API_URL.format(
-                    self.item_type, self.organization), auth=auth)
-        if repository_data.status_code == requests.codes.NOT_FOUND:
-            lgr.error(
+    def _get_org_data(self):
+        response = requests.get(constants.GITHUB_API_URL.format(
+            self.item_type, self.organization), auth=self.git_credentials)
+        if response.status_code == requests.codes.NOT_FOUND:
+            self.logger.error(
                 'The organization or user {0} could not be found. '
                 'Please make sure you use the correct type (org/user).'.format(
                     self.organization))
             sys.exit(1)
-        repository_number = \
-            repository_data.json()['{0}_repos'.format(repository_type)]
-        last_page_number = repository_number / repository_per_page
-        if (repository_number % repository_per_page) > 0:
+        return response.json()
+
+    def _get_repos_list(self, repos_per_page, page_num):
+        response = requests.get(constants.REPO_DETAILS_API_URL.format(
+            self.item_type,
+            self.organization,
+            'public',
+            repos_per_page,
+            page_num), auth=self.git_credentials)
+        return response.json()
+
+    def _parse_repo_data(self, repo_data):
+        return [dict((key, data[key]) for key in ['name', 'clone_url'])
+                for data in repo_data]
+
+    def _get_repos_data(self, repos_per_page=100):
+        self.logger.info(
+            'Retrieving repository information for the {0}...'.format(
```

Comment:
nice to have - if self._organization then also log the organization else log the user name.